// @histoire/app/dist/bundled/components/search/SearchModal.vue_vue_type_script_setup_true_lang.js
import { defineComponent, defineAsyncComponent, withDirectives, openBlock, createElementBlock, createElementVNode, createVNode, unref, vShow } from "@histoire/vendors/vue";
import "./SearchLoading.js";
import _sfc_main$1 from "./SearchLoading.vue_vue_type_script_setup_true_lang.js";
const _hoisted_1 = {
  class: "histoire-search-modal htw-fixed htw-inset-0 htw-bg-white/80 dark:htw-bg-gray-700/80 htw-z-20",
  "data-test-id": "search-modal"
};
const _hoisted_2 = { class: "htw-bg-white dark:htw-bg-gray-700 md:htw-mt-16 md:htw-mx-auto htw-w-screen htw-max-w-[512px] htw-shadow-xl htw-border htw-border-gray-200 dark:htw-border-gray-850 htw-rounded-lg htw-relative htw-divide-y htw-divide-gray-200 dark:htw-divide-gray-850" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "SearchModal",
  props: {
    shown: {
      type: Boolean,
      default: false
    }
  },
  emits: {
    close: () => true
  },
  setup(__props, { emit }) {
    const SearchPane = defineAsyncComponent({
      loader: () => import("./SearchPane.js"),
      loadingComponent: _sfc_main$1,
      delay: 0
    });
    function close() {
      emit("close");
    }
    return (_ctx, _cache) => {
      return withDirectives((openBlock(), createElementBlock("div", _hoisted_1, [
        createElementVNode("div", {
          class: "htw-absolute htw-inset-0",
          onClick: _cache[0] || (_cache[0] = ($event) => close())
        }),
        createElementVNode("div", _hoisted_2, [
          createVNode(unref(SearchPane), {
            shown: __props.shown,
            onClose: _cache[1] || (_cache[1] = ($event) => close())
          }, null, 8, ["shown"])
        ])
      ], 512)), [
        [vShow, __props.shown]
      ]);
    };
  }
});
export {
  _sfc_main as default
};
