// @histoire/plugin-vue/dist/bundled/client/server/run.js
import { createApp as c, h as f } from "vue";
import * as n from "virtual:$histoire-setup";
import * as r from "virtual:$histoire-generated-global-setup";
import s from "./Story.js";
import d from "./Variant.js";
async function v({ file: a, storyData: u, el: i }) {
  const { default: e } = await import(
    /* @vite-ignore */
    a.moduleId
  ), t = c({
    provide: {
      addStory(o) {
        u.push(o);
      }
    },
    render() {
      return f(e, {
        ref: "comp",
        data: a
      });
    }
  });
  if (t.component("Story", s), t.component("Variant", d), typeof (r == null ? void 0 : r.setupVue3) == "function" && await r.setupVue3({
    app: t,
    story: null,
    variant: null
  }), typeof (n == null ? void 0 : n.setupVue3) == "function" && await n.setupVue3({
    app: t,
    story: null,
    variant: null
  }), t.mount(i), e.doc) {
    const o = document.createElement("div");
    o.innerHTML = e.doc;
    const p = o.textContent;
    u.forEach((m) => {
      m.docsText = p;
    });
  }
  t.unmount();
}
export {
  v as run
};
