// @histoire/app/dist/bundled/components/story/StoryVariantListItem.vue_vue_type_script_setup_true_lang.js
import { defineComponent, useCssVars, unref, toRefs, ref, openBlock, createElementBlock, createVNode, withCtx, normalizeClass, createElementVNode, toDisplayString, pushScopeId, popScopeId } from "@histoire/vendors/vue";
import { useCurrentVariantRoute } from "../../util/variant.js";
import BaseListItemLink from "../base/BaseListItemLink.js";
import { Icon } from "@histoire/vendors/iconify";
import { useScrollOnActive } from "../../util/scroll.js";
const _withScopeId = (n) => (pushScopeId("data-v-ab890d9c"), n = n(), popScopeId(), n);
const _hoisted_1 = { class: "htw-truncate" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "StoryVariantListItem",
  props: {
    variant: {
      type: Object,
      required: true
    }
  },
  setup(__props) {
    const props = __props;
    useCssVars((_ctx) => ({
      "89126aaa": unref(variant).iconColor
    }));
    const { variant } = toRefs(props);
    const { isActive, targetRoute } = useCurrentVariantRoute(variant);
    const el = ref();
    useScrollOnActive(isActive, el);
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        ref_key: "el",
        ref: el,
        class: "histoire-story-variant-list-item",
        "data-test-id": "story-variant-list-item"
      }, [
        createVNode(BaseListItemLink, {
          to: unref(targetRoute),
          "is-active": unref(isActive),
          class: "htw-px-2 htw-py-2 md:htw-py-1.5 htw-m-1 htw-rounded-sm htw-flex htw-items-center htw-gap-2"
        }, {
          default: withCtx(({ active }) => {
            var _a;
            return [
              createVNode(unref(Icon), {
                icon: (_a = unref(variant).icon) != null ? _a : "carbon:cube",
                class: normalizeClass(["htw-w-5 htw-h-5 sm:htw-w-4 sm:htw-h-4 htw-flex-none", {
                  "htw-text-gray-500": !active && !unref(variant).iconColor,
                  "bind-icon-color": !active && unref(variant).iconColor
                }])
              }, null, 8, ["icon", "class"]),
              createElementVNode("span", _hoisted_1, toDisplayString(unref(variant).title), 1)
            ];
          }),
          _: 1
        }, 8, ["to", "is-active"])
      ], 512);
    };
  }
});
export {
  _sfc_main as default
};
