// @histoire/app/dist/bundled/components/story/StoryVariantSinglePreviewNative.vue_vue_type_script_setup_true_lang.js
import { defineComponent, computed, openBlock, createBlock, withCtx, createElementVNode, normalizeStyle, unref, normalizeClass } from "@histoire/vendors/vue";
import { isDark } from "../../util/dark.js";
import { histoireConfig } from "../../util/config.js";
import { usePreviewSettingsStore } from "../../stores/preview-settings.js";
import { getContrastColor } from "../../util/preview-settings.js";
import "./GenericRenderStory.js";
import StoryResponsivePreview from "./StoryResponsivePreview.js";
import _sfc_main$1 from "./GenericRenderStory.vue_vue_type_script_setup_true_lang.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "StoryVariantSinglePreviewNative",
  props: {
    story: null,
    variant: null
  },
  setup(__props) {
    const props = __props;
    Object.assign(props.variant, {
      previewReady: false
    });
    function onReady() {
      Object.assign(props.variant, {
        previewReady: true
      });
    }
    const settings = usePreviewSettingsStore().currentSettings;
    const contrastColor = computed(() => getContrastColor(settings));
    const autoApplyContrastColor = computed(() => !!histoireConfig.autoApplyContrastColor);
    return (_ctx, _cache) => {
      return openBlock(), createBlock(StoryResponsivePreview, {
        class: "histoire-story-variant-single-preview-native",
        variant: __props.variant
      }, {
        default: withCtx(({ isResponsiveEnabled, finalWidth, finalHeight }) => [
          createElementVNode("div", {
            style: normalizeStyle([
              isResponsiveEnabled ? {
                width: finalWidth ? `${finalWidth}px` : "100%",
                height: finalHeight ? `${finalHeight}px` : "100%"
              } : { width: "100%", height: "100%" },
              {
                "--histoire-contrast-color": unref(contrastColor),
                color: unref(autoApplyContrastColor) ? unref(contrastColor) : void 0
              }
            ]),
            class: "htw-relative",
            "data-test-id": "sandbox-render"
          }, [
            (openBlock(), createBlock(_sfc_main$1, {
              key: `${__props.story.id}-${__props.variant.id}`,
              variant: __props.variant,
              story: __props.story,
              class: normalizeClass(["htw-h-full", [unref(isDark) ? unref(histoireConfig).sandboxDarkClass : void 0]]),
              dir: unref(settings).textDirection,
              onReady
            }, null, 8, ["variant", "story", "class", "dir"]))
          ], 4)
        ]),
        _: 1
      }, 8, ["variant"]);
    };
  }
});
export {
  _sfc_main as default
};
