// @histoire/app/dist/bundled/components/tree/StoryGroup.vue_vue_type_script_setup_true_lang.js
import { createElementVNode, defineComponent, computed, openBlock, createElementBlock, withKeys, createVNode, unref, normalizeClass, toDisplayString, createCommentVNode, Fragment, renderList, createBlock } from "@histoire/vendors/vue";
import StoryListItem from "./StoryListItem.js";
import StoryListFolder from "./StoryListFolder.js";
import { Icon } from "@histoire/vendors/iconify";
import { useFolderStore } from "../../stores/folder.js";
const _hoisted_1 = {
  "data-test-id": "story-group",
  class: "histoire-story-group htw-my-2 first:htw-mt-0 last:htw-mb-0"
};
const _hoisted_2 = ["onKeyup"];
const _hoisted_3 = { class: "htw-truncate htw-opacity-50" };
const _hoisted_4 = /* @__PURE__ */ createElementVNode("span", { class: "htw-h-[1px] htw-flex-1 htw-bg-gray-500/10 htw-mx-2" }, null, -1);
const _hoisted_5 = { key: 1 };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "StoryGroup",
  props: {
    path: { default: () => [] },
    group: null,
    stories: null
  },
  setup(__props) {
    const props = __props;
    const folderStore = useFolderStore();
    const folderPath = computed(() => [...props.path, props.group.title]);
    const isFolderOpen = computed(() => folderStore.isFolderOpened(folderPath.value, true));
    function toggleOpen() {
      folderStore.toggleFolder(folderPath.value, false);
    }
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", _hoisted_1, [
        __props.group.title ? (openBlock(), createElementBlock("div", {
          key: 0,
          role: "button",
          tabindex: "0",
          class: "htw-px-0.5 htw-py-2 md:htw-py-1.5 htw-mx-1 htw-rounded-sm hover:htw-bg-primary-100 dark:hover:htw-bg-primary-900 htw-cursor-pointer htw-select-none htw-flex htw-items-center htw-gap-1 htw-min-w-0",
          onClick: toggleOpen,
          onKeyup: [
            withKeys(toggleOpen, ["enter"]),
            withKeys(toggleOpen, ["space"])
          ]
        }, [
          createVNode(unref(Icon), {
            icon: "carbon:caret-right",
            class: normalizeClass(["htw-w-4 htw-h-4 htw-transition-transform htw-duration-150 htw-opacity-30 htw-flex-none", {
              "htw-rotate-90": unref(isFolderOpen)
            }])
          }, null, 8, ["class"]),
          createElementVNode("span", _hoisted_3, toDisplayString(__props.group.title), 1),
          _hoisted_4
        ], 40, _hoisted_2)) : createCommentVNode("", true),
        unref(isFolderOpen) ? (openBlock(), createElementBlock("div", _hoisted_5, [
          (openBlock(true), createElementBlock(Fragment, null, renderList(__props.group.children, (element) => {
            return openBlock(), createElementBlock(Fragment, {
              key: element.title
            }, [
              element.children ? (openBlock(), createBlock(StoryListFolder, {
                key: 0,
                path: unref(folderPath),
                folder: element,
                stories: __props.stories,
                depth: 0
              }, null, 8, ["path", "folder", "stories"])) : (openBlock(), createBlock(StoryListItem, {
                key: 1,
                story: __props.stories[element.index],
                depth: 0
              }, null, 8, ["story"]))
            ], 64);
          }), 128))
        ])) : createCommentVNode("", true)
      ]);
    };
  }
});
export {
  _sfc_main as default
};
