// @histoire/app/dist/bundled/components/tree/StoryListFolder.vue_vue_type_script_setup_true_lang.js
import { pushScopeId, popScopeId, defineComponent, useCssVars, unref, computed, resolveComponent, openBlock, createElementBlock, createElementVNode, withKeys, createVNode, normalizeClass, toDisplayString, Fragment, renderList, createBlock, createCommentVNode } from "@histoire/vendors/vue";
import StoryListItem from "./StoryListItem.js";
import { Icon } from "@histoire/vendors/iconify";
import { useFolderStore } from "../../stores/folder.js";
const _withScopeId = (n) => (pushScopeId("data-v-2fedbe45"), n = n(), popScopeId(), n);
const _hoisted_1 = {
  "data-test-id": "story-list-folder",
  class: "histoire-story-list-folder"
};
const _hoisted_2 = ["onKeyup"];
const _hoisted_3 = { class: "bind-tree-padding htw-flex htw-items-center htw-gap-2 htw-min-w-0" };
const _hoisted_4 = { class: "htw-flex htw-flex-none htw-items-center" };
const _hoisted_5 = { class: "htw-truncate" };
const _hoisted_6 = { key: 0 };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "StoryListFolder",
  props: {
    path: { default: () => [] },
    folder: null,
    stories: null,
    depth: { default: 0 }
  },
  setup(__props) {
    const props = __props;
    useCssVars((_ctx) => ({
      "29654456": unref(folderPadding)
    }));
    const folderStore = useFolderStore();
    const folderPath = computed(() => [...props.path, props.folder.title]);
    const isFolderOpen = computed(() => folderStore.isFolderOpened(folderPath.value));
    function toggleOpen() {
      folderStore.toggleFolder(folderPath.value);
    }
    const folderPadding = computed(() => {
      return props.depth * 12 + "px";
    });
    return (_ctx, _cache) => {
      const _component_StoryListFolder = resolveComponent("StoryListFolder", true);
      return openBlock(), createElementBlock("div", _hoisted_1, [
        createElementVNode("div", {
          role: "button",
          tabindex: "0",
          class: "htw-px-0.5 htw-py-2 md:htw-py-1.5 htw-mx-1 htw-rounded-sm hover:htw-bg-primary-100 dark:hover:htw-bg-primary-900 htw-cursor-pointer htw-select-none htw-flex",
          onClick: toggleOpen,
          onKeyup: [
            withKeys(toggleOpen, ["enter"]),
            withKeys(toggleOpen, ["space"])
          ]
        }, [
          createElementVNode("span", _hoisted_3, [
            createElementVNode("span", _hoisted_4, [
              createVNode(unref(Icon), {
                icon: "carbon:caret-right",
                class: normalizeClass(["htw-w-4 htw-h-4 htw-transition-transform htw-duration-150 htw-opacity-30", {
                  "htw-rotate-90": unref(isFolderOpen)
                }])
              }, null, 8, ["class"]),
              createVNode(unref(Icon), {
                icon: "carbon:folder",
                class: "htw-w-5 htw-h-5 sm:htw-w-4 sm:htw-h-4"
              })
            ]),
            createElementVNode("span", _hoisted_5, toDisplayString(__props.folder.title), 1)
          ])
        ], 40, _hoisted_2),
        unref(isFolderOpen) ? (openBlock(), createElementBlock("div", _hoisted_6, [
          (openBlock(true), createElementBlock(Fragment, null, renderList(__props.folder.children, (element) => {
            return openBlock(), createElementBlock(Fragment, {
              key: element.title
            }, [
              element.children ? (openBlock(), createBlock(_component_StoryListFolder, {
                key: 0,
                path: unref(folderPath),
                folder: element,
                stories: __props.stories,
                depth: __props.depth + 1
              }, null, 8, ["path", "folder", "stories", "depth"])) : (openBlock(), createBlock(StoryListItem, {
                key: 1,
                story: __props.stories[element.index],
                depth: __props.depth + 1
              }, null, 8, ["story", "depth"]))
            ], 64);
          }), 128))
        ])) : createCommentVNode("", true)
      ]);
    };
  }
});
export {
  _sfc_main as default
};
