// @histoire/app/dist/bundled/components/tree/StoryListItem.vue_vue_type_script_setup_true_lang.js
import { pushScopeId, popScopeId, defineComponent, useCssVars, unref, computed, ref, openBlock, createElementBlock, createVNode, withCtx, createElementVNode, normalizeClass, toDisplayString, createCommentVNode } from "@histoire/vendors/vue";
import BaseListItemLink from "../base/BaseListItemLink.js";
import { Icon } from "@histoire/vendors/iconify";
import { useRoute } from "@histoire/vendors/vue-router";
import { useScrollOnActive } from "../../util/scroll.js";
const _withScopeId = (n) => (pushScopeId("data-v-19810958"), n = n(), popScopeId(), n);
const _hoisted_1 = { class: "bind-tree-margin htw-flex htw-items-center htw-gap-2 htw-pl-4 htw-min-w-0" };
const _hoisted_2 = { class: "htw-truncate" };
const _hoisted_3 = {
  key: 0,
  class: "htw-opacity-40 htw-text-sm"
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "StoryListItem",
  props: {
    story: null,
    depth: { default: 0 }
  },
  setup(__props) {
    const props = __props;
    useCssVars((_ctx) => ({
      "444efc12": unref(filePadding),
      "ae892164": __props.story.iconColor
    }));
    const filePadding = computed(() => {
      return props.depth * 12 + "px";
    });
    const route = useRoute();
    const isActive = computed(() => route.params.storyId === props.story.id);
    const el = ref();
    useScrollOnActive(isActive, el);
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        ref_key: "el",
        ref: el,
        "data-test-id": "story-list-item",
        class: "histoire-story-list-item"
      }, [
        createVNode(BaseListItemLink, {
          to: {
            name: "story",
            params: {
              storyId: __props.story.id
            }
          },
          class: "htw-pl-0.5 htw-pr-2 htw-py-2 md:htw-py-1.5 htw-mx-1 htw-rounded-sm"
        }, {
          default: withCtx(({ active }) => {
            var _a;
            return [
              createElementVNode("span", _hoisted_1, [
                createVNode(unref(Icon), {
                  icon: (_a = __props.story.icon) != null ? _a : "carbon:cube",
                  class: normalizeClass(["htw-w-5 htw-h-5 sm:htw-w-4 sm:htw-h-4 htw-flex-none", {
                    "htw-text-primary-500": !active && !__props.story.iconColor,
                    "bind-icon-color": !active && __props.story.iconColor
                  }])
                }, null, 8, ["icon", "class"]),
                createElementVNode("span", _hoisted_2, toDisplayString(__props.story.title), 1)
              ]),
              !__props.story.docsOnly ? (openBlock(), createElementBlock("span", _hoisted_3, toDisplayString(__props.story.variants.length), 1)) : createCommentVNode("", true)
            ];
          }),
          _: 1
        }, 8, ["to"])
      ], 512);
    };
  }
});
export {
  _sfc_main as default
};
