// @histoire/app/dist/bundled/components/panel/ControlsComponentPropItem.vue_vue_type_script_setup_true_lang.js
import { defineComponent, computed, resolveDirective, unref, openBlock, createBlock, resolveDynamicComponent, isRef, withCtx, withDirectives, createVNode, normalizeClass, withModifiers, createCommentVNode } from "@histoire/vendors/vue";
import { Icon } from "@histoire/vendors/iconify";
import { HstJson, HstCheckbox, HstNumber, HstText } from "@histoire/controls";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ControlsComponentPropItem",
  props: {
    variant: null,
    component: null,
    definition: null
  },
  setup(__props) {
    const props = __props;
    const comp = computed(() => {
      var _a;
      switch ((_a = props.definition.types) == null ? void 0 : _a[0]) {
        case "string":
          return HstText;
        case "number":
          return HstNumber;
        case "boolean":
          return HstCheckbox;
        case "object":
        default:
          return HstJson;
      }
    });
    const model = computed({
      get: () => {
        var _a;
        return (_a = props.variant.state._hPropState[props.component.index]) == null ? void 0 : _a[props.definition.name];
      },
      set: (value) => {
        if (!props.variant.state._hPropState[props.component.index]) {
          props.variant.state._hPropState[props.component.index] = {};
        }
        props.variant.state._hPropState[props.component.index][props.definition.name] = value;
      }
    });
    function reset() {
      if (props.variant.state._hPropState[props.component.index]) {
        delete props.variant.state._hPropState[props.component.index][props.definition.name];
      }
    }
    const canReset = computed(() => {
      var _a, _b;
      return ((_b = (_a = props.variant.state) == null ? void 0 : _a._hPropState) == null ? void 0 : _b[props.component.index]) && props.definition.name in props.variant.state._hPropState[props.component.index];
    });
    return (_ctx, _cache) => {
      const _directive_tooltip = resolveDirective("tooltip");
      return unref(comp) ? (openBlock(), createBlock(resolveDynamicComponent(unref(comp)), {
        key: 0,
        modelValue: unref(model),
        "onUpdate:modelValue": _cache[1] || (_cache[1] = ($event) => isRef(model) ? model.value = $event : null),
        class: "histoire-controls-component-prop-item",
        title: `${__props.definition.name}${unref(canReset) ? " *" : ""}`
      }, {
        actions: withCtx(() => [
          withDirectives(createVNode(unref(Icon), {
            icon: "carbon:erase",
            class: normalizeClass(["htw-cursor-pointer htw-w-4 htw-h-4 hover:htw-text-primary-500 dark:hover:htw-text-primary-400 htw-text-gray-900 dark:htw-text-gray-100", [
              unref(canReset) ? "htw-opacity-50 hover:htw-opacity-100" : "htw-opacity-25 htw-pointer-events-none"
            ]]),
            onClick: _cache[0] || (_cache[0] = withModifiers(($event) => reset(), ["stop"]))
          }, null, 8, ["class"]), [
            [_directive_tooltip, "Remove override"]
          ])
        ]),
        _: 1
      }, 8, ["modelValue", "title"])) : createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as default
};
