// @histoire/plugin-vue/dist/bundled/client/app/RenderStory.js
import { reactive as _, createApp as N, onMounted as R } from "vue";
import { defineComponent as V, ref as x, onMounted as j, watch as T, onBeforeUnmount as k, h as q } from "@histoire/vendors/vue";
import { applyState as w } from "@histoire/shared";
import { getTagName as O } from "../codegen.js";
import { registerGlobalComponents as B } from "./global-components.js";
import { RouterLinkStub as C } from "./RouterLinkStub.js";
import * as l from "virtual:$histoire-setup";
import * as m from "virtual:$histoire-generated-global-setup";
import { syncStateBundledAndExternal as D } from "./util.js";
const z = V({
  name: "RenderStory",
  props: {
    variant: {
      type: Object,
      required: !0
    },
    story: {
      type: Object,
      required: !0
    },
    slotName: {
      type: String,
      default: "default"
    }
  },
  emits: {
    ready: () => !0
  },
  setup(t, { emit: P }) {
    const v = x();
    let r, d = !1;
    const o = _({});
    D(t.variant.state, o);
    function h() {
      r && (r.unmount(), r = null);
    }
    async function b() {
      if (d)
        return;
      d = !0, h();
      let i;
      r = N({
        name: "RenderStorySubApp",
        setup() {
          R(() => {
            d = !1;
          });
        },
        render: () => {
          var u, c, e, p, a;
          const s = (a = (c = (u = t.variant.slots()) == null ? void 0 : u[t.slotName]) == null ? void 0 : c.call(u, {
            state: o
          })) != null ? a : (p = (e = t.story.slots()) == null ? void 0 : e[t.slotName]) == null ? void 0 : p.call(e, {
            state: o
          });
          if (t.slotName === "default" && !t.variant.autoPropsDisabled) {
            const n = S(s), y = JSON.stringify(n);
            (!i || i !== y) && (w(t.variant.state, {
              _hPropDefs: n
            }), t.variant.state._hPropState || w(t.variant.state, {
              _hPropState: {}
            }), i = y);
          }
          return s;
        }
      }), B(r), r.component("RouterLink", C), typeof (m == null ? void 0 : m.setupVue3) == "function" && await m.setupVue3({
        app: r,
        story: t.story,
        variant: t.variant
      }), typeof (l == null ? void 0 : l.setupVue3) == "function" && await l.setupVue3({
        app: r,
        story: t.story,
        variant: t.variant
      }), typeof t.variant.setupApp == "function" && await t.variant.setupApp({
        app: r,
        story: t.story,
        variant: t.variant
      });
      const f = document.createElement("div");
      v.value.appendChild(f), r.mount(f), P("ready");
    }
    function S(i) {
      var u, c;
      const f = [];
      let s = 0;
      for (const e of i) {
        if (typeof e.type == "object") {
          const p = [];
          for (const a in e.type.props) {
            const n = e.type.props[a];
            let y, g;
            n && (y = (Array.isArray(n.type) ? n.type : typeof n == "function" ? [n] : [n.type]).map((A) => {
              switch (A) {
                case String:
                  return "string";
                case Number:
                  return "number";
                case Boolean:
                  return "boolean";
                case Object:
                  return "object";
                case Array:
                  return "array";
                default:
                  return "unknown";
              }
            }), g = typeof n.default == "function" ? n.default.toString() : n.default), p.push({
              name: a,
              types: y,
              required: n == null ? void 0 : n.required,
              default: g
            }), ((c = (u = o == null ? void 0 : o._hPropState) == null ? void 0 : u[s]) == null ? void 0 : c[a]) != null && (e.props || (e.props = {}), e.props[a] = o._hPropState[s][a], e.dynamicProps || (e.dynamicProps = []), e.dynamicProps.includes(a) || e.dynamicProps.push(a));
          }
          f.push({
            name: O(e),
            index: s,
            props: p
          }), s++;
        }
        Array.isArray(e.children) && f.push(...S(e.children));
      }
      return f.filter((e) => e.props.length);
    }
    return j(async () => {
      t.variant.configReady && await b();
    }), T(() => t.variant, async (i) => {
      i.configReady && !d && (r ? r._instance.proxy.$forceUpdate() : await b());
    }, {
      deep: !0
    }), k(() => {
      h();
    }), {
      sandbox: v
    };
  },
  render() {
    return q("div", {
      ref: "sandbox",
      class: "__histoire-sandbox htw-overflow-auto"
    });
  }
});
export {
  z as default
};
