// @histoire/plugin-vue/dist/bundled/client/app/Variant.js
import { defineComponent as o, useAttrs as r, getCurrentInstance as u, inject as l } from "vue";
import { applyState as p } from "@histoire/shared";
import { toRawDeep as c, syncStateBundledAndExternal as d } from "./util.js";
const v = o({
  name: "Variant",
  __histoireType: "variant",
  props: {
    initState: {
      type: Function,
      default: null
    },
    source: {
      type: String,
      default: null
    },
    responsiveDisabled: {
      type: Boolean,
      default: !1
    },
    autoPropsDisabled: {
      type: Boolean,
      default: !1
    },
    setupApp: {
      type: Function,
      default: null
    }
  },
  async setup(t) {
    const e = r(), n = u(), i = l("implicitState");
    if (typeof t.initState == "function") {
      const s = await t.initState();
      p(e.variant.state, c(s));
    }
    d(e.variant.state, i());
    function a() {
      Object.assign(e.variant, {
        slots: () => n.proxy.$slots,
        source: t.source,
        responsiveDisabled: t.responsiveDisabled,
        autoPropsDisabled: t.autoPropsDisabled,
        setupApp: t.setupApp,
        configReady: !0
      });
    }
    return a(), {
      updateVariant: a
    };
  },
  render() {
    return this.updateVariant(), null;
  }
});
export {
  v as default
};
